containers/unit_tests/TestDynViewAPI.hpp

```diff
@@ -1402,10 +1402,7 @@ class TestDynViewAPI {
     // For CUDA the constant random access View does not return
     // an lvalue reference due to retrieving through texture cache
     // therefore not allowed to query the underlying pointer.
-#if defined(KOKKOS_ENABLE_CUDA)
-    if (!std::is_same<typename device::execution_space, Kokkos::Cuda>::value)
-#endif
-    {
+    if (!std::is_same<typename device::execution_space, Kokkos::Cuda>::value) {
```

Comment:
Non-blocking comment: won't this be a problem for, e.g., HIP also? Shouldn't the `data()` method be SFINAE'd out if this isn't an allowed operation?

Reply:
I created an issue for that: https://github.com/kokkos/kokkos/issues/3894